sdk/iot/Azure.Iot.Hub.Service/src/IoTHubServiceClient.cs

```diff
@@ -81,5 +106,68 @@ public IoTHubServiceClient(Uri endpoint, IoTHubServiceClientOptions options)
             Files = new FilesClient();
             Jobs = new JobsClient();
         }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IoTHubServiceClient"/> class.
+        /// </summary>
+        /// <param name="endpoint">
+        /// The IoT Hub service instance URI to connect to.
+        /// </param>
+        /// <param name="credential">
+        /// The <see cref="TokenCredential"/> implementation which will be used to request for the authentication token.
+        /// </param>
+        public IoTHubServiceClient(Uri endpoint, TokenCredential credential)
+            : this(endpoint, credential, new IoTHubServiceClientOptions())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IoTHubServiceClient"/> class.
+        /// </summary>
+        /// <param name="endpoint">
+        /// The IoT Hub service instance URI to connect to.
+        /// </param>
+        /// <param name="credential">
+        /// The <see cref="TokenCredential"/> implementation which will be used to request for the authentication token.
+        /// </param>
+        /// <param name="options">
+        /// Options that allow configuration of requests sent to the IoT Hub service.
+        /// </param>
+        public IoTHubServiceClient(Uri endpoint, TokenCredential credential, IoTHubServiceClientOptions options)
```

Comment:
This is a pending TODO: verify if IoT Hub supports OAuth token based authentication

Reply:
Don't add this until we confirm that this is supported.

Reply:
I haven't added this block. It was previously added, probably as a placeholder so that we remember to verify TokenCredential support. 
Even if OAuth based support is available in IoT Hub, we would still need to have sas token based support as well (unless IoT Hub is planning on deprecating that support).
Do you feel otherwise?

Reply:
We need to have that discussion before adding the code is what I meant :) We need to evaluate what we need to do. Why don't we do a design review first? I don't think we reviewed this with Andrei yet.

Reply:
Tracked here: https://msazure.visualstudio.com/One/_workitems/edit/7443468